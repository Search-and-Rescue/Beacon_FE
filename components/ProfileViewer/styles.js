const styles = {
  profileContainer: {
    flex: 1,
    marginHorizontal: '10%',
    marginVertical: 40,
    width: '80%'
  }
}

export default styles;
