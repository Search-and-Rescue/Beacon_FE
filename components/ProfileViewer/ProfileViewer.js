import React from "react";
import { View, Text } from "react-native";
import styles from "./styles";
import { connect } from "react-redux";

const ProfileViewer = props => {
  const {
    address,
    allergies,
    birthDate,
    city,
    cosarCard,
    email,
    experienceLevel,
    gender,
    hairColor,
    height,
    name,
    passwordDigest,
    phone,
    skinColor,
    state,
    weight,
    zip
  } = props.user;

  return (
    <View style={styles.profileContainer}>
      <Text style={styles.label}>Name: {name}</Text>

      <Text style={styles.label}>Home address: {address}</Text>
      <Text style={styles.label}>City: {city}</Text>
      <Text style={styles.label}>State: {state}</Text>
      <Text style={styles.label}>Zip code: {zip}</Text>
      <Text style={styles.label}>Phone number: {phone}</Text>
      <Text style={styles.label}>Email address: {email}</Text>

      <Text style={styles.label}>Birthday: {birthDate}</Text>
      <Text style={styles.label}>Password: {passwordDigest}</Text>

      <Text style={styles.label}>Height: {weight}</Text>
      <Text style={styles.label}>Weight: {height}</Text>
      <Text style={styles.label}>Hair Color: {hairColor}</Text>
      <Text style={styles.label}>Skin Color: {skinColor}</Text>
      <Text style={styles.label}>Gender: {gender}</Text>

      <Text style={styles.label}>Allergies: {allergies}</Text>
      <Text style={styles.modalToggleText}>Outdoor Experience Level: {experienceLevel}</Text>
      <Text style={styles.modalToggleText}>COSAR Card Status: {cosarCard}</Text>
    </View>
  );
};

export const mapStateToProps = ({ user }) => ({
  user
});

export default connect(mapStateToProps)(ProfileViewer);
