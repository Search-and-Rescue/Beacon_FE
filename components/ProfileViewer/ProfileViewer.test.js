import "react-native";
import React from "react";
import { shallow } from "enzyme";
import { ProfileViewer } from "./ProfileViewer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let mockUser, wrapper;

beforeEach(() => {
  mockUser = {
    address: "123 Main St.",
    allergies: "Juniper Berries",
    birthDate: "1985-05-01",
    city: "Leadville",
    cosarCard: true,
    email: "ohhello@email.com",
    experienceLevel: 3,
    gender: "F",
    hairColor: "silver",
    height: "509",
    name: "Ms. Hunter Dickinson",
    phone: "(710)123-4567",
    skinColor: "olive",
    state: "CO",
    weight: "150",
    zip: "80202"
  };

  wrapper = shallow(<ProfileViewer user={mockUser} />);
})

describe("ProfileViewer", () => {
  it("should match the snapshot with all of the data passed through", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
