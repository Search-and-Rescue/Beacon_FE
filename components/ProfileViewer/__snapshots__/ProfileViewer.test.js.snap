// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileViewer should match the snapshot with all of the data passed through 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F0F0",
      "borderColor": "#001028",
      "borderWidth": 4,
      "flex": 1,
      "marginHorizontal": "5%",
      "marginTop": 20,
      "width": "90%",
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#001028",
        "color": "#F0F0F0",
        "fontFamily": "Futura",
        "fontSize": 21,
        "marginBottom": 7,
        "paddingLeft": 5,
        "paddingVertical": 2,
      }
    }
  >
    About Me: 
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Ms. Hunter Dickinson
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    123 Main St.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Leadville
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    CO
    , 
    80202
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    (710)123-4567
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    ohhello@email.com
  </Text>
  <Text
    style={
      Object {
        "backgroundColor": "#001028",
        "color": "#F0F0F0",
        "fontFamily": "Futura",
        "fontSize": 21,
        "marginBottom": 7,
        "paddingLeft": 5,
        "paddingVertical": 2,
      }
    }
  >
    Personal Information: 
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Birthday: 
    1985-05-01
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Password: *********
  </Text>
  <Text
    style={
      Object {
        "backgroundColor": "#001028",
        "color": "#F0F0F0",
        "fontFamily": "Futura",
        "fontSize": 21,
        "marginBottom": 7,
        "paddingLeft": 5,
        "paddingVertical": 2,
      }
    }
  >
    Physical Description: 
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Height: 
    150
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Weight: 
    509
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Hair Color: 
    silver
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Skin Color: 
    olive
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Gender: 
    F
  </Text>
  <Text
    style={
      Object {
        "backgroundColor": "#001028",
        "color": "#F0F0F0",
        "fontFamily": "Futura",
        "fontSize": 21,
        "marginBottom": 7,
        "paddingLeft": 5,
        "paddingVertical": 2,
      }
    }
  >
    First Responder FYI: 
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Allergies: 
    Juniper Berries
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    Outdoor Experience Level: 
    3
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingBottom": 2,
        "paddingLeft": 5,
      }
    }
  >
    COSAR Card Status: 
    Current
  </Text>
</View>
`;
